### ISSUE TITLE
Android unreachable emulator after fresh install

### ISSUE THREAD

**Issue by LHLK** [NativeScript/nativescript-cli#3244]
I followed your document in this URL: https://docs.nativescript.org/angular/tutorial/ng-chapter-1#11-install-nativescript-and-configure-your-environment 
(1) The installation procedures was outdated and not working.  This script:
@powershell -NoProfile -ExecutionPolicy Bypass -Command "iex ((new-object net.webclient).DownloadString('https://www.nativescript.org/setup/win'))"  from your document was not working; BUT it did installed most of the thing like the Android SDK version 23 and I believe the tns files too. That's why the tns doctor told me "no errors" !
(2) I'd managed to solve the tns device problem by (1) removing the above script installed SDK (2) re-installed the SDK by installing the latest Android Studio.
(3) Now if I ran the tns run android, the emulator show up but with the following errors: "Unable to apply changes on device: emulator-5554. Error is: The device is not authorized. Please use the --emulator flag to run the application on on an emulator. "
(4) I google this errors for few hours and I did try everything like adb kill-sever restart-server and the status of the tns devices never changed as following: 
$ tns devices
Connected devices & emulators
Searching for devices...
iTunes is not installed. Install it on your system and run this command again.
┌───┬─────────────┬──────────┬───────────────────┬────────┬─────────────┐
│ # │ Device Name │ Platform │ Device Identifier │ Type   │ Status      │
│ 1 │             │ Android  │ emulator-5554     │ Device │ Unreachable │
└───┴─────────────┴──────────┴───────────────────┴────────┴─────────────┘

(5) My simple logic is, if the tns run Android is able to launch the emulator.  Why it's unreachable? 

To be honest, I'm pretty disappointed to NativeScript as it's just the VERY FIRST STEP to explore your goodness.  But failed!!!

**Comment by NickIliev** [comment]
@LHLK `tns devices` will list the devices but still, they have to be connected and working via ADB in order to deploy and run any Android applications (no matter native Android or NativeScript ones).
You can run the respective Android command to test the connectivity to your emulator with
```
adb devices
```

and it should list similar output if the emulator is connected and working
```
List of devices attached
emulator-5554   device
```

For some emulators, the error `"The device is not authorized"` is related to enabling USB debugging via the Android settings (discussed as a solution [here](https://github.com/NativeScript/nativescript-cli/issues/1882#issuecomment-314822266)). Try this solution to solve your case (just like you would do on a real device to enable USB debugging in the `Developer options`). If you don't see the developer options menu in your `Settings` use [this guide](https://www.greenbot.com/article/2457986/android/how-to-enable-developer-options-on-your-android-phone-or-tablet.html) to enable it.

For other a UI dialog should be accepted on the first run of the emulator.

**Comment by LHLK** [comment]
Thanks for your prompt reply.  Here is what I did to find out the problem:
(1) I removed the nativeScript package using the command "npm uninstall -g nativeScript" entirely.
(2) I followed exactly the same steps with your URL: https://docs.nativescript.org/angular/start/quick-setup 
(3) I executed the provide script "@powershell -NoProfile -ExecutionPolicy Bypass -Command "iex ((new-object net.webclient).DownloadString('https://www.nativescript.org/setup/win'))"
(4) Verified with tns doctor with no erros.
(5) I followed your URL:  "https://docs.nativescript.org/angular/tutorial/ng-chapter-1#11-install-nativescript-and-configure-your-environment" and entered the statement: "tns create HelloWorld --template nativescript-template-ng-tutorial"
(6) I connected my Android (real) device with the USB (just to by-pass the emulator issue). 
(7) Then cd HelloWorld and tns run Android.  
(8) Note: The installation was success and I saw the HelloWorld icon appeared in the Android device. However, the following errors came out: 
Running full build
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Project successfully built.
Installing...
Successfully installed on device with identifier 'faf6dda'.
Successfully transferred all files.
Refreshing application...
Successfully synced application org.nativescript.HelloWorld on device faf6dda.
Executing before-watch hook from E:\Project\Training\learnNativeScript\HelloWorld\hooks\before-watch\nativescript-dev-typescript.js
Found peer TypeScript 2.4.2
ActivityManager: Start proc org.nativescript.HelloWorld for activity org.nativescript.HelloWorld/com.tns.NativeScriptActivity: pid=8891 uid=10195 gids={50195, 1028, 1015, 3003}
16:09:58 - Compilation complete. Watching for file changes.



System.err: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.nativescript.HelloWorld/com.tns.NativeScriptActivity}: com.tns.NativeScriptException:
System.err: Calling js method onCreate failed
System.err: TypeError: activityInfo.metaData.getInt is not a function
System.err: File: "file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/ui/frame/frame.js, line: 701, column: 18
System.err: StackTrace:
System.err:     Frame: function:'', file:'file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/application/application.js', line: 144, column: 62
System.err:     Frame: function:'ActivityCallbacksImplementation.onCreate', file:'file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 701, column: 19
System.err:     Frame: function:'NativeScriptActivity.onCreate', file:'file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/ui/frame/activity.js', line: 20, column: 25
System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2200)
System.err:     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2249)
System.err:     at android.app.ActivityThread.access$800(ActivityThread.java:141)
System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1212)
System.err:     at android.os.Handler.dispatchMessage(Handler.java:102)
System.err:     at android.os.Looper.loop(Looper.java:136)
System.err:     at android.app.ActivityThread.main(ActivityThread.java:5052)
System.err:     at java.lang.reflect.Method.invokeNative(Native Method)
System.err:     at java.lang.reflect.Method.invoke(Method.java:515)
System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:796)
System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:612)
System.err:     at dalvik.system.NativeStart.main(Native Method)
System.err: Caused by: com.tns.NativeScriptException:
System.err: Calling js method onCreate failed
System.err: TypeError: activityInfo.metaData.getInt is not a function
System.err: File: "file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/ui/frame/frame.js, line: 701, column: 18
System.err: StackTrace:
System.err:     Frame: function:'', file:'file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/application/application.js', line: 144, column: 62
System.err:     Frame: function:'ActivityCallbacksImplementation.onCreate', file:'file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/ui/frame/frame.js', line: 701, column: 19
System.err:     Frame: function:'NativeScriptActivity.onCreate', file:'file:///data/data/org.nativescript.HelloWorld/files/app/tns_modules/tns-core-modules/ui/frame/activity.js', line: 20, column: 25
System.err:     at com.tns.Runtime.callJSMethodNative(Native Method)
System.err:     at com.tns.Runtime.dispatchCallJSMethodNative(Runtime.java:1088)
System.err:     at com.tns.Runtime.callJSMethodImpl(Runtime.java:970)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:957)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:941)
System.err:     at com.tns.Runtime.callJSMethod(Runtime.java:933)
System.err:     at com.tns.NativeScriptActivity.onCreate(NativeScriptActivity.java:13)
System.err:     at android.app.Activity.performCreate(Activity.java:5242)
System.err:     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087)
System.err:     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2164)
System.err:     ... 11 more
ActivityManager: Process org.nativescript.HelloWorld (pid 8891) has died.
ActivityManager: cleanUpApplicationRecordLocked -- 8891

END of the error.

(9) There was also error show up in the Android device as well.  
Please help.  I really don't know how to solve the problem as I simply followed the steps stated from your tutorial.

**Comment by NickIliev** [comment]
@ligaz are you by chance trying to deploy on Xiaomi Readmi device? Your issue might be related to [this one](https://github.com/NativeScript/android-runtime/issues/404)  (some Xiaomi and other devices are using their own custom implementation of Android SDK and are not compatible with the official SDK used by NativeScript). However, deploying on emulator should not be an issue - have you tried the solution provided in my previous comment?

**Comment by LHLK** [comment]
Yes sir. I'm using the Xiaomi so it's confirmed that this type of phone won't work with NativeScript? I'll try to work on the emulator problem.  Thanks.

**Comment by NickIliev** [comment]
Yes, it is confirmed that some Xiaomi devices (and rooted phones) are incompatible with NativeScript due to the fact that they are not using the official Android SDK but their own overwritten SDK.

Do let me know if you need further assistance with enabling the emulator on your side,

**Comment by LHLK** [comment]
Got it. I'll work on the emulator and update back you.  Ah, just couple of my point of views and for your information, this Xiaomi Android phones worked fine during my training course on the Android Studio Development about two years ago.  And I did just tested with the Hello World apps with Ionic 3 without any error.  Thanks again for your prompt response.   Very good indeed.

**Comment by LHLK** [comment]
The emulator issue had been solved. Thank you NickIliev.  I created another VM for testing this time and all of your scripts in your tutorial works; however, I will suggest to add the followings: 
(1) If the user didn't install the Android Studio, after following the current tutorial doc the user (like me) will have a hard time to install the virtual devices.  So it's suggested to inform the user, without installing the Android Studio, they can type (if the path is correct) android avd and run the android.bat batch file (Windows 10) under the android SDK directory. 
(2) Then the user can import and create the virtual devices as the above two commands will launch the SDK manager and AVD manager? in order to complete the emulator.
(3) Start the emulator and confirmed with adb devices.  
Thank you.
